### ISSUE TITLE
dub hangs indefinitely when it can't find json/sdl package file

### ISSUE THREAD

**Issue by pszturmaj** [dlang/dub#1678]
Wanted to try D after years of not using it. Downloaded dexed v3.7.9 and dmd 2.085.1. I'm on Win10 x64.

Following hello world code is compiling right now for few minutes:
``` d
module runnable;

import std.stdio;

void main(string[] args)
{
    writeln("Test");
}
```
Above code is placed in source/app.d. Dub takes 100% CPU in task manager and 2.8 MB of RAM. Windows Defender is disabled. I still have DMD 2.052 on disk and I remember there were some similar issues with dub few years ago. Compiling 'hello world' in 2019 for 10 minutes and still not seeing the results may seriously repel D language newcomers.
![dub_slow](https://user-images.githubusercontent.com/178614/55987344-a210a280-5ca1-11e9-9339-9ee25dec9aeb.PNG)
[In case u wonder why there is 15% CPU on the screenshot - I have 8 cores and it's actually using more than one core]
I tried running dub manually but it seems the same so I terminated it after 1 minute.

**Comment by wilzbach** [comment]
This is very weird. Maybe an Anti-Virus program is blocking dub?
Do you see the same issues with just plain dmd?
Do you have any project dependencies?

**Comment by pszturmaj** [comment]
Windows Defender is disabled. There is no other Anti-Virus software installed.
I just ran dmd on app.d. In half a second it compiled and linked fresh executable (which works).
Zero dependencies, only above "hello world" source code.

**Comment by andre2007** [comment]
Could you try 2 things:
- enable very verbose output for dub with --vverbose
- remove the module statement in app.d

Also what is the content of dub.json?

**Comment by pszturmaj** [comment]
```
D:\Projects\test_dlang_new>dub --verbose
Using dub registry url 'https://code.dlang.org/'
Refreshing local packages (refresh existing: true)...
Looking for local package map at C:\ProgramData\dub\packages\local-packages.json
Looking for local package map at C:\Users\Piotr\AppData\Local\dub\packages\local-packages.json
No valid package found in current working directory: No package file found in D:\Projects\test_dlang_new\, expected one of dub.json/dub.sdl/package.json
There was no package description found for the application in 'D:\Projects\test_dlang_new'.
Note: Failed to determine version of package  at .. Assuming ~master.
Refreshing local packages (refresh existing: false)...
Looking for local package map at C:\ProgramData\dub\packages\local-packages.json
Looking for local package map at C:\Users\Piotr\AppData\Local\dub\packages\local-packages.json
Refreshing local packages (refresh existing: false)...
Looking for local package map at C:\ProgramData\dub\packages\local-packages.json
Looking for local package map at C:\Users\Piotr\AppData\Local\dub\packages\local-packages.json
Warning: The package in D:/Projects/test_dlang_new/ has no name.
```

Dub json file was created automatically by dexed. Contents:
```
{
  "name" : ""
}
```

**Comment by andre2007** [comment]
What happens when you set a name in dub.json?

**Comment by pszturmaj** [comment]
Tried with few names and removed the UTF8 BOM from the json file, still hangs.

**Comment by andre2007** [comment]
Where is dub.json is located? I ask because of this log line: 
No package file found in D:\Projects\test_dlang_new\, expected one of dub.json/dub.sdl/package.json

**Comment by pszturmaj** [comment]
You are right, thanks. I named the json file differently (I forgot it should be dub/package.json). Now it compiles quickly. However I still think it's a bug, dub should terminate when reporting such an error, not hang indefinitely.

This may help catching it:
```
D:\Projects\test_dlang_new>dub --vverbose
Using dub registry url 'https://code.dlang.org/'
Refreshing local packages (refresh existing: true)...
Looking for local package map at C:\ProgramData\dub\packages\local-packages.json
Looking for local package map at C:\Users\Piotr\AppData\Local\dub\packages\local-packages.json
No valid package found in current working directory: No package file found in D:\Projects\test_dlang_new\, expected one of dub.json/dub.sdl/package.json
There was no package description found for the application in 'D:\Projects\test_dlang_new'.
Note: Failed to determine version of package  at .. Assuming ~master.
Refreshing local packages (refresh existing: false)...
Looking for local package map at C:\ProgramData\dub\packages\local-packages.json
Looking for local package map at C:\Users\Piotr\AppData\Local\dub\packages\local-packages.json
Collecting dependencies for
Refreshing local packages (refresh existing: false)...
Looking for local package map at C:\ProgramData\dub\packages\local-packages.json
Looking for local package map at C:\Users\Piotr\AppData\Local\dub\packages\local-packages.json
Collecting dependencies for
Warning: The package in D:/Projects/test_dlang_new/ has no name.
Add config  application
Add config  library
NON-PRIMARY:  library
Eliminating config library for
NON-PRIMARY:
Eliminating config  for
NON-PRIMARY:
Eliminating config  for
NON-PRIMARY:
Eliminating config  for
NON-PRIMARY:
Eliminating config  for
NON-PRIMARY:
Eliminating config  for
NON-PRIMARY:
Eliminating config  for
NON-PRIMARY:
Eliminating config  for
NON-PRIMARY:
Eliminating config  for
NON-PRIMARY:
Eliminating config  for
NON-PRIMARY:
Eliminating config  for
NON-PRIMARY:
Eliminating config  for
NON-PRIMARY:
Eliminating config  for
NON-PRIMARY:
Eliminating config  for
NON-PRIMARY:
Eliminating config  for
NON-PRIMARY:
Eliminating config  for
NON-PRIMARY:

[* infinite stream of previous 2 lines *]
```

**Comment by andre2007** [comment]
Thanks you are right, it should not hang.  For your info, please name it dub.json. package.json is more or less deprecated.

**Comment by Geod24** [comment]
Duplicate of #1277, will be fixed in dub 1.20.0.
